### PR TITLE
Fix tests on linux.

### PR DESCRIPTION
**Pull request by omeshp**

### CODE REVIEW

File: Extensions/ArtifactEngine/Engine/artifactEngine.ts [hunk 1]
```diff
@@ -85,7 +85,22 @@ export class ArtifactEngine {
...
         if (item.itemType === models.ItemType.File) {
-            if (tl.match([item.path], this.patternList).length > 0) {
+            var pathToMatch = item.path.replace(/\\/g, '/');
+            var matchOptions = {
+                debug: false,
+                nobrace: true,
+                noglobstar: false,
+                dot: true,
+                noext: false,
+                nocase: true,
```
Comment (GitHubSriramB) [inline Extensions/ArtifactEngine/Engine/artifactEngine.ts]: does this mean case sensitive?

File: Extensions/ArtifactEngine/Engine/artifactEngine.ts [hunk 2]
```diff
@@ -85,7 +85,22 @@ export class ArtifactEngine {
...
+                nonull: false,
+                matchBase: false,
+                nocomment: false,
+                nonegate: false,
+                flipNegate: false
+            };
+
+            if (tl.match([pathToMatch], this.patternList, null, matchOptions).length > 0) {
```
Comment (GitHubSriramB) [inline Extensions/ArtifactEngine/Engine/artifactEngine.ts]: don't we need to fix patternList to also not have back-slashes?

File: Extensions/ArtifactEngine/EngineTests/artifactEngineTests.ts [hunk 3]
```diff
@@ -80,7 +96,7 @@ describe('artifactEngine.processItems', () => {
     it('should call getArtifactItem only for artifact items that match include pattern', (done) => {
         var testProvider = new providers.StubProvider();
         var downloadOptions = new engine.ArtifactEngineOptions();
-        downloadOptions.itemPattern = 'path1\\**\npath3\\**\n!path4\\**';
+        downloadOptions.itemPattern = 'path1/**\npath3/**\n!path4/**';
```
Comment (GitHubSriramB) [inline Extensions/ArtifactEngine/EngineTests/artifactEngineTests.ts]: keep one test that has back-slash in pattern & make sure it continues to work?

Comment (omeshp) [reply Extensions/ArtifactEngine/EngineTests/artifactEngineTests.ts]: already there.

File: Extensions/ArtifactEngine/Engine/artifactEngine.ts (hunk 2 above)
Comment (omeshp) [reply Extensions/ArtifactEngine/Engine/artifactEngine.ts]: not required.

File: Extensions/ArtifactEngine/Engine/artifactEngine.ts (hunk 1 above)
Comment (omeshp) [reply Extensions/ArtifactEngine/Engine/artifactEngine.ts]: yes, this should be false (case insensitive), will change

**Review by GitHubSriramB — APPROVED** [review]

**Pull request merged**